src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreSettings.cs

```diff
@@ -17,5 +17,37 @@ public class ProjectRestoreSettings
         /// Currently this is only being used for net core based projects on nomination.
         /// </summary>
         public bool HideWarningsAndErrors { get; set; } = false;
+
+        public ProjectRestoreSettings Clone()
+        {
+            var clonedObject = new ProjectRestoreSettings();
+            clonedObject.HideWarningsAndErrors = HideWarningsAndErrors;
+            return clonedObject;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as ProjectRestoreSettings);
+        }
+
+        public bool Equals(ProjectRestoreSettings other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return HideWarningsAndErrors == other.HideWarningsAndErrors;
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```

Comment:
This should hash `HideWarningsAndErrors`

Reply:
I autogened that...guess it I didn't include the field by accident. 

Updated it with another autogened hashcode.